pkg/utils/node/predicates.go

```diff
@@ -54,3 +55,9 @@ func IsUnderutilized(node *v1.Node, pods []*v1.Pod) bool {
 	}
 	return true
 }
+
+func IsKarpenterProvisioned(node *v1.Node) bool {
```

Comment:
I don't think we want an awareness of the provisioner labels. More explicit would be to check the finalizer. At the end of the day, think of this controller as a node finalizer that is completely decoupled from provisioning concepts.

Reply:
Removed it.